Add swagger implementation

vicayala82 opened the pull request:
## Summary
-Integrated swagger functionality
-Created documentation for User Controller
## Trello Card
https://trello.com/c/3itYz35c/40-swagger

Commit:
project base created, firth commit card  https://trello.com/c/9Ai3dnqx/42-pull-request

Commit:
modify .gitnore whit gradle folder

Commit:
greeting view and controller book developed card https://trello.com/c/Wz8y9wmp/17-crear-un-controller-y-una-vista-de-bienvenida

Commit:
created Book model class for card https://trello.com/c/WmDWOCFU/14-crear-modelo-de-book

Commit:
Create bookRepository interface card https://trello.com/c/yNYUtYVk/15-crear-un-repositorio-de-books

Commit:
-created crud methods in controller class
-new customer exception classes
-methods get and setter for Book.id
-New search for title
card https://trello.com/b/jz5sf0BB/training-victor-ayala

Commit:
-documented code with javadocs
card https://trello.com/c/PycY2Bx1/47-javadoc

Commit:
-Created User model class
-Created methods for add and remove book of collection
-Created model relationship user whit book
-Created BookAlreadyOwnedException class
card https://trello.com/c/PycY2Bx1/47-javadoc

Commit:
-Created UserRepository Interface
card https://trello.com/c/0kM2SNoy/20-crear-un-repositorio-de-users

Commit:
-Created UserCrontroller class
-Created Exceptions UserIdMismatchException and UserNotFoundException
-Created CRUD methods for User controlled
-Update some book logic
card https://trello.com/c/SeUuyKTK/21-crear-user-controller-con-las-operaciones-crud-de-user

Commit:
-Integrated swagger functionality
-Created documentation for User Controller
card https://trello.com/c/3itYz35c/40-swagger

juandbc commented on training/src/main/java/wolox/training/models/User.java:
The `@ ApiModelProperty` annotation for the other attributes are missing. Also, add the `required` argument to the annotation

vicayala82 committed:
Update training/src/main/java/wolox/training/repositoies/BookRepository.java

Thanks for the recommendation :)

Co-authored-by: Juan David Bermudez Celedon <juanbermucele@gmail.com>

vicayala82 committed:
Update training/src/main/java/wolox/training/models/Book.java

OK, thanks

Co-authored-by: Juan David Bermudez Celedon <juanbermucele@gmail.com>

vicayala82 committed:
Update training/src/main/java/wolox/training/controllers/BookController.java

Co-authored-by: Juan David Bermudez Celedon <juanbermucele@gmail.com>

Commit:
merge resolve conflicts

vicayala82 committed:
Update training/src/main/java/wolox/training/exceptions/BookIdMismatchException.java

Ok

Co-authored-by: Juan David Bermudez Celedon <juanbermucele@gmail.com>

vicayala82 committed:
Update training/src/main/java/wolox/training/exceptions/BookIdMismatchException.java

Thanks

Co-authored-by: Juan David Bermudez Celedon <juanbermucele@gmail.com>

vicayala82 committed:
Update training/src/main/java/wolox/training/exceptions/RestExceptionHandler.java

Understood, thanks

Co-authored-by: Juan David Bermudez Celedon <juanbermucele@gmail.com>

vicayala82 committed:
Update training/src/main/java/wolox/training/exceptions/BookNotFoundException.java

ok

Co-authored-by: Juan David Bermudez Celedon <juanbermucele@gmail.com>

Commit:
Merge branch 'create-book-crud-operation' of https://github.com/vicayala82/vda-java into create-book-crud-operation

Commit:
-Fixed with recomendations
card https://trello.com/c/jez5TteF/18-crear-las-operaciones-crud-de-book

vicayala82 committed:
Update training/src/main/java/wolox/training/models/User.java

Co-authored-by: Juan David Bermudez Celedon <juanbermucele@gmail.com>

vicayala82 committed:
Update training/src/main/java/wolox/training/models/User.java

Co-authored-by: Juan David Bermudez Celedon <juanbermucele@gmail.com>

vicayala82 committed:
Update training/src/main/java/wolox/training/models/User.java

Ok, thanks

Co-authored-by: Juan David Bermudez Celedon <juanbermucele@gmail.com>

Commit:
Resolved conflict in Merge an fixed recomendations

Commit:
Merge branch 'create-user-model-class' of https://github.com/vicayala82/vda-java into create-user-model-class

Commit:
-Fixed with recomendations
-merge and resolve conflicts
card https://trello.com/c/2fqH75mu/19-crear-modelo-de-user

Commit:
Merge branch 'create-user-model-class' into create-user-repository-class

vicayala82 committed:
Update training/src/main/java/wolox/training/repositoies/UserRepository.java

Co-authored-by: Juan David Bermudez Celedon <juanbermucele@gmail.com>

Commit:
Merge branch 'create-user-repository-class' of https://github.com/vicayala82/vda-java into create-user-repository-class

Commit:
-Fixed with recomendations
-merge and resolve conflicts
card https://trello.com/c/0kM2SNoy/20-crear-un-repositorio-de-users

vicayala82 committed:
Update training/src/main/java/wolox/training/controllers/UserController.java

Understood, thanks

Co-authored-by: Juan David Bermudez Celedon <juanbermucele@gmail.com>

vicayala82 committed:
Update training/src/main/java/wolox/training/controllers/UserController.java

Co-authored-by: Juan David Bermudez Celedon <juanbermucele@gmail.com>

Commit:
Resolved Merge conflicts

Commit:
-Fixed with recomendations
-merge and resolve conflicts
-fixed exceptions
card https://trello.com/c/SeUuyKTK/21-crear-user-controller-con-las-operaciones-crud-de-user

Commit:
Resolved Merge conflicts

vicayala82 commented on training/src/main/java/wolox/training/models/User.java:
Fixed annotations in User.java

Commit:
-Fixed with recomendations
-merge and resolve conflicts
card https://trello.com/c/3itYz35c/40-swagger
